solo-chains/runtime/starlight/src/tests/collator_assignment_tests.rs

```diff
@@ -714,8 +735,22 @@ fn test_paras_registered_but_zero_credits() {
             assert_ok!(ContainerRegistrar::register(
                 origin_of(ALICE.into()),
                 1001.into(),
-                empty_genesis_data()
+                get_genesis_data_with_validation_code().0,
+                Some(HeadData(vec![1u8, 1u8, 1u8]))
+            ));
+
+            run_to_session(2);
+            let assignment = TanssiCollatorAssignment::collator_container_chain();
+            assert!(assignment.container_chains.is_empty());
+
+            // We need to accept the validation code, so that the para is onboarded after 2 sessions.
+            assert_ok!(Paras::add_trusted_validation_code(
+                root_origin(),
+                get_genesis_data_with_validation_code().1.into()
             ));
+            run_to_session(4);
+            let assignment = TanssiCollatorAssignment::collator_container_chain();
+            assert!(assignment.container_chains.is_empty());
```

Comment:
I am curious to see, what happens if we register a container-chain that has an non-valid wasm? does it get collators assigned after 2 sessions?

I guess it's no problem, as it will pay for those so there is a clear disincentive for that, but can we add a test? meaning we register but we never add the `trusted_validation_code`

Reply:
Sure, I added a test in cf97156e020b5066cefd0db772b3913ff3e98966. 

The result shows that no collators are assigned to the container chain. This is because if we don't mark the validation code as trusted, when we call `mark_valid_for_collating`, it will simply fail with `NotParathread` error (as the paraId has not yet been onboarded and thus `mark_valid_for_collating` cannot upgrade it to a parachain in the relay context).

Reply:
fixed